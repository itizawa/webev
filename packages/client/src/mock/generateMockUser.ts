import { User } from '@monorepo/webev-client/src/domains/User';

export const generateMockUser = (mock: Partial<User> = {}): User => {
  return new User({
    _id: mock._id || 'mockId',
    name: mock.name || 'mockName',
    description: mock.description || 'mockDescription',
    email: mock.email || 'mockEmail',
    image: mock.image || 'mockImage',
    admin: mock.admin || false,
    isExecutedTutorial: mock.isExecutedTutorial || true,
    apiTokenForExtension: mock.apiTokenForExtension || 'mockApiTokenForExtension',
    createdAt: mock.createdAt || new Date('2020-01-01T00:00:00'),
    updatedAt: mock.updatedAt || new Date('2020-01-01T00:00:00'),
  });
};
