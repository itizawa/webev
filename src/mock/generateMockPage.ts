import { Page } from '~/domains/Page';

export const generateMockPage = (mock: Partial<Page> = {}): Page => {
  return new Page({
    _id: mock._id || 'mockId',
    url: mock.url || 'https://example.com',
    image: mock.image || 'mockImage',
    favicon: mock.favicon || 'mockFavicon',
    description: mock.description || 'mockDescription',
    title: mock.title || 'mockTitle',
    body: mock.body || 'mockBody',
    siteName: mock.siteName || 'mockSiteName',
    isDeleted: mock.isDeleted || false,
    createdUser: mock.createdUser || 'mockCreatedUser',
    createdAt: mock.createdAt || new Date('2020-01-01T00:00:00'),
    updatedAt: mock.updatedAt || new Date('2020-01-01T00:00:00'),
    archivedAt: mock.archivedAt || new Date('2020-01-01T00:00:00'),
  });
};
