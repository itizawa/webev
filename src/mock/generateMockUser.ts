import { User } from '~/domains/User';

export const generateMockUser = (mock: Partial<User> = {}): User => {
  return new User({
    _id: mock._id || 'mockId',
    name: mock.name || 'mockName',
    description: mock.description || 'mockDescription',
    email: mock.email || 'mockEmail',
    image: mock.image || 'mockImage',
    admin: mock.admin || false,
    isExecutedTutorial: mock.isExecutedTutorial || true,
    apiTokenForExtension: mock.apiTokenForExtension || 'mockApiTokenForExtension',
    createdAt: mock.createdAt || new Date(),
    updatedAt: mock.updatedAt || new Date(),
  });
};
