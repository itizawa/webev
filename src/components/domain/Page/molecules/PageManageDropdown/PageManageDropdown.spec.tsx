import renderer from 'react-test-renderer';
import { PageManageDropdown } from './index';
import { generateMockPage } from '~/mock/generateMockPage';

const mockPage = generateMockPage();

test('PageManageDropdown isHideArchiveButton is true', () => {
  const tree = renderer.create(<PageManageDropdown page={mockPage} onClickFetchButton={() => console.log('test')} />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('PageManageDropdown directoryId is undefined', () => {
  const tree = renderer.create(<PageManageDropdown page={mockPage} onClickFetchButton={() => console.log('test')} />).toJSON();

  expect(tree).toMatchSnapshot();
});

global.navigator = {
  ...global.navigator,
  share: jest.fn(),
};

test('PageManageDropdown directoryId is undefined', () => {
  const tree = renderer.create(<PageManageDropdown page={mockPage} onClickFetchButton={() => console.log('test')} />).toJSON();

  expect(tree).toMatchSnapshot();
});
