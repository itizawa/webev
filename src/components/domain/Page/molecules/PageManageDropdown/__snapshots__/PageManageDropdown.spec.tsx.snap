// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageManageDropdown directoryId is undefined 1`] = `
<div
  className="dropdown dropstart"
  onKeyDown={[Function]}
>
  <span
    aria-expanded={false}
    aria-haspopup={true}
    className=""
    onClick={[Function]}
  >
    <div
      id="manage-for-mockId"
    >
      <button
        className="btn px-2 border-0"
        onClick={[Function]}
      >
        <svg
          className="align-text-bottom text-white"
          fill="currentColor"
          height={18}
          viewBox="0 0 16 16"
          width={18}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.5 13a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0z"
          />
        </svg>
      </button>
    </div>
  </span>
  <div
    aria-hidden={true}
    className="dropdown-menu-dark border-secondary dropdown-menu"
    positionFixed={true}
    role="menu"
    tabIndex="-1"
  >
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      <svg
        className="align-text-bottom text-white"
        fill="currentColor"
        height={18}
        viewBox="0 0 16 16"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 7a.5.5 0 0 1 .5.5V9H10a.5.5 0 0 1 0 1H8.5v1.5a.5.5 0 0 1-1 0V10H6a.5.5 0 0 1 0-1h1.5V7.5A.5.5 0 0 1 8 7z"
          fillRule="evenodd"
        />
        <path
          d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"
        />
        <path
          d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"
        />
      </svg>
      <span
        className="ms-2"
      >
        URL をコピーする
      </span>
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      <svg
        className="align-text-bottom text-white"
        fill="currentColor"
        height={18}
        viewBox="0 0 16 16"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.5 5.5A.5.5 0 0 1 6 6v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm2.5 0a.5.5 0 0 1 .5.5v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm3 .5a.5.5 0 0 0-1 0v6a.5.5 0 0 0 1 0V6z"
        />
        <path
          d="M14.5 3a1 1 0 0 1-1 1H13v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V4h-.5a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1H6a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1h3.5a1 1 0 0 1 1 1v1zM4.118 4L4 4.059V13a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4.059L11.882 4H4.118zM2.5 3V2h11v1h-11z"
          fillRule="evenodd"
        />
      </svg>
      <span
        className="ms-2"
      >
        削除
      </span>
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      <svg
        className="align-text-bottom text-white"
        fill="currentColor"
        height={18}
        viewBox="0 0 16 16"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 3a5 5 0 1 0 4.546 2.914.5.5 0 0 1 .908-.417A6 6 0 1 1 8 2v1z"
          fillRule="evenodd"
        />
        <path
          d="M8 4.466V.534a.25.25 0 0 1 .41-.192l2.36 1.966c.12.1.12.284 0 .384L8.41 4.658A.25.25 0 0 1 8 4.466z"
        />
      </svg>
      <span
        className="ms-2"
      >
        ページを再取得する
      </span>
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      <svg
        className="align-text-bottom text-white"
        fill="currentColor"
        height={18}
        viewBox="0 0 16 16"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13.5 1a1.5 1.5 0 1 0 0 3 1.5 1.5 0 0 0 0-3zM11 2.5a2.5 2.5 0 1 1 .603 1.628l-6.718 3.12a2.499 2.499 0 0 1 0 1.504l6.718 3.12a2.5 2.5 0 1 1-.488.876l-6.718-3.12a2.5 2.5 0 1 1 0-3.256l6.718-3.12A2.5 2.5 0 0 1 11 2.5zm-8.5 4a1.5 1.5 0 1 0 0 3 1.5 1.5 0 0 0 0-3zm11 5.5a1.5 1.5 0 1 0 0 3 1.5 1.5 0 0 0 0-3z"
        />
      </svg>
      <span
        className="ms-2"
      >
        共有
      </span>
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      <svg
        className="align-text-bottom text-white"
        fill="currentColor"
        height={20}
        viewBox="0 0 16 16"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.921 11.9 1.353 8.62a.719.719 0 0 1 0-1.238L5.921 4.1A.716.716 0 0 1 7 4.719V6c1.5 0 6 0 7 8-2.5-4.5-7-4-7-4v1.281c0 .56-.606.898-1.079.62z"
        />
      </svg>
      <span
        className="ms-2 text-nowrap"
      >
        戻す
      </span>
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      <svg
        className="align-text-bottom text-white"
        fill="currentColor"
        height={18}
        viewBox="0 0 16 16"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m.5 3 .04.87a1.99 1.99 0 0 0-.342 1.311l.637 7A2 2 0 0 0 2.826 14H9v-1H2.826a1 1 0 0 1-.995-.91l-.637-7A1 1 0 0 1 2.19 4h11.62a1 1 0 0 1 .996 1.09L14.54 8h1.005l.256-2.819A2 2 0 0 0 13.81 3H9.828a2 2 0 0 1-1.414-.586l-.828-.828A2 2 0 0 0 6.172 1H2.5a2 2 0 0 0-2 2zm5.672-1a1 1 0 0 1 .707.293L7.586 3H2.19c-.24 0-.47.042-.683.12L1.5 2.98a1 1 0 0 1 1-.98h3.672z"
        />
        <path
          d="M13.5 10a.5.5 0 0 1 .5.5V12h1.5a.5.5 0 1 1 0 1H14v1.5a.5.5 0 1 1-1 0V13h-1.5a.5.5 0 0 1 0-1H13v-1.5a.5.5 0 0 1 .5-.5z"
        />
      </svg>
      <span
        className="ms-2"
      >
        ディレクトリにページを追加する
      </span>
    </button>
  </div>
</div>
`;

exports[`PageManageDropdown directoryId is undefined 2`] = `
<div
  className="dropdown dropstart"
  onKeyDown={[Function]}
>
  <span
    aria-expanded={false}
    aria-haspopup={true}
    className=""
    onClick={[Function]}
  >
    <div
      id="manage-for-mockId"
    >
      <button
        className="btn px-2 border-0"
        onClick={[Function]}
      >
        <svg
          className="align-text-bottom text-white"
          fill="currentColor"
          height={18}
          viewBox="0 0 16 16"
          width={18}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.5 13a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0z"
          />
        </svg>
      </button>
    </div>
  </span>
  <div
    aria-hidden={true}
    className="dropdown-menu-dark border-secondary dropdown-menu"
    positionFixed={true}
    role="menu"
    tabIndex="-1"
  >
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      <svg
        className="align-text-bottom text-white"
        fill="currentColor"
        height={18}
        viewBox="0 0 16 16"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 7a.5.5 0 0 1 .5.5V9H10a.5.5 0 0 1 0 1H8.5v1.5a.5.5 0 0 1-1 0V10H6a.5.5 0 0 1 0-1h1.5V7.5A.5.5 0 0 1 8 7z"
          fillRule="evenodd"
        />
        <path
          d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"
        />
        <path
          d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"
        />
      </svg>
      <span
        className="ms-2"
      >
        URL をコピーする
      </span>
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      <svg
        className="align-text-bottom text-white"
        fill="currentColor"
        height={18}
        viewBox="0 0 16 16"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.5 5.5A.5.5 0 0 1 6 6v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm2.5 0a.5.5 0 0 1 .5.5v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm3 .5a.5.5 0 0 0-1 0v6a.5.5 0 0 0 1 0V6z"
        />
        <path
          d="M14.5 3a1 1 0 0 1-1 1H13v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V4h-.5a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1H6a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1h3.5a1 1 0 0 1 1 1v1zM4.118 4L4 4.059V13a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4.059L11.882 4H4.118zM2.5 3V2h11v1h-11z"
          fillRule="evenodd"
        />
      </svg>
      <span
        className="ms-2"
      >
        削除
      </span>
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      <svg
        className="align-text-bottom text-white"
        fill="currentColor"
        height={18}
        viewBox="0 0 16 16"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 3a5 5 0 1 0 4.546 2.914.5.5 0 0 1 .908-.417A6 6 0 1 1 8 2v1z"
          fillRule="evenodd"
        />
        <path
          d="M8 4.466V.534a.25.25 0 0 1 .41-.192l2.36 1.966c.12.1.12.284 0 .384L8.41 4.658A.25.25 0 0 1 8 4.466z"
        />
      </svg>
      <span
        className="ms-2"
      >
        ページを再取得する
      </span>
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      <svg
        className="align-text-bottom text-white"
        fill="currentColor"
        height={18}
        viewBox="0 0 16 16"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13.5 1a1.5 1.5 0 1 0 0 3 1.5 1.5 0 0 0 0-3zM11 2.5a2.5 2.5 0 1 1 .603 1.628l-6.718 3.12a2.499 2.499 0 0 1 0 1.504l6.718 3.12a2.5 2.5 0 1 1-.488.876l-6.718-3.12a2.5 2.5 0 1 1 0-3.256l6.718-3.12A2.5 2.5 0 0 1 11 2.5zm-8.5 4a1.5 1.5 0 1 0 0 3 1.5 1.5 0 0 0 0-3zm11 5.5a1.5 1.5 0 1 0 0 3 1.5 1.5 0 0 0 0-3z"
        />
      </svg>
      <span
        className="ms-2"
      >
        共有
      </span>
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      <svg
        className="align-text-bottom text-white"
        fill="currentColor"
        height={20}
        viewBox="0 0 16 16"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.921 11.9 1.353 8.62a.719.719 0 0 1 0-1.238L5.921 4.1A.716.716 0 0 1 7 4.719V6c1.5 0 6 0 7 8-2.5-4.5-7-4-7-4v1.281c0 .56-.606.898-1.079.62z"
        />
      </svg>
      <span
        className="ms-2 text-nowrap"
      >
        戻す
      </span>
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      <svg
        className="align-text-bottom text-white"
        fill="currentColor"
        height={18}
        viewBox="0 0 16 16"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m.5 3 .04.87a1.99 1.99 0 0 0-.342 1.311l.637 7A2 2 0 0 0 2.826 14H9v-1H2.826a1 1 0 0 1-.995-.91l-.637-7A1 1 0 0 1 2.19 4h11.62a1 1 0 0 1 .996 1.09L14.54 8h1.005l.256-2.819A2 2 0 0 0 13.81 3H9.828a2 2 0 0 1-1.414-.586l-.828-.828A2 2 0 0 0 6.172 1H2.5a2 2 0 0 0-2 2zm5.672-1a1 1 0 0 1 .707.293L7.586 3H2.19c-.24 0-.47.042-.683.12L1.5 2.98a1 1 0 0 1 1-.98h3.672z"
        />
        <path
          d="M13.5 10a.5.5 0 0 1 .5.5V12h1.5a.5.5 0 1 1 0 1H14v1.5a.5.5 0 1 1-1 0V13h-1.5a.5.5 0 0 1 0-1H13v-1.5a.5.5 0 0 1 .5-.5z"
        />
      </svg>
      <span
        className="ms-2"
      >
        ディレクトリにページを追加する
      </span>
    </button>
  </div>
</div>
`;

exports[`PageManageDropdown isHideArchiveButton is true 1`] = `
<div
  className="dropdown dropstart"
  onKeyDown={[Function]}
>
  <span
    aria-expanded={false}
    aria-haspopup={true}
    className=""
    onClick={[Function]}
  >
    <div
      id="manage-for-mockId"
    >
      <button
        className="btn px-2 border-0"
        onClick={[Function]}
      >
        <svg
          className="align-text-bottom text-white"
          fill="currentColor"
          height={18}
          viewBox="0 0 16 16"
          width={18}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.5 13a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0z"
          />
        </svg>
      </button>
    </div>
  </span>
  <div
    aria-hidden={true}
    className="dropdown-menu-dark border-secondary dropdown-menu"
    positionFixed={true}
    role="menu"
    tabIndex="-1"
  >
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      <svg
        className="align-text-bottom text-white"
        fill="currentColor"
        height={18}
        viewBox="0 0 16 16"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 7a.5.5 0 0 1 .5.5V9H10a.5.5 0 0 1 0 1H8.5v1.5a.5.5 0 0 1-1 0V10H6a.5.5 0 0 1 0-1h1.5V7.5A.5.5 0 0 1 8 7z"
          fillRule="evenodd"
        />
        <path
          d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"
        />
        <path
          d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"
        />
      </svg>
      <span
        className="ms-2"
      >
        URL をコピーする
      </span>
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      <svg
        className="align-text-bottom text-white"
        fill="currentColor"
        height={18}
        viewBox="0 0 16 16"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.5 5.5A.5.5 0 0 1 6 6v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm2.5 0a.5.5 0 0 1 .5.5v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm3 .5a.5.5 0 0 0-1 0v6a.5.5 0 0 0 1 0V6z"
        />
        <path
          d="M14.5 3a1 1 0 0 1-1 1H13v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V4h-.5a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1H6a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1h3.5a1 1 0 0 1 1 1v1zM4.118 4L4 4.059V13a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4.059L11.882 4H4.118zM2.5 3V2h11v1h-11z"
          fillRule="evenodd"
        />
      </svg>
      <span
        className="ms-2"
      >
        削除
      </span>
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      <svg
        className="align-text-bottom text-white"
        fill="currentColor"
        height={18}
        viewBox="0 0 16 16"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 3a5 5 0 1 0 4.546 2.914.5.5 0 0 1 .908-.417A6 6 0 1 1 8 2v1z"
          fillRule="evenodd"
        />
        <path
          d="M8 4.466V.534a.25.25 0 0 1 .41-.192l2.36 1.966c.12.1.12.284 0 .384L8.41 4.658A.25.25 0 0 1 8 4.466z"
        />
      </svg>
      <span
        className="ms-2"
      >
        ページを再取得する
      </span>
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      <svg
        className="align-text-bottom text-white"
        fill="currentColor"
        height={18}
        viewBox="0 0 16 16"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13.5 1a1.5 1.5 0 1 0 0 3 1.5 1.5 0 0 0 0-3zM11 2.5a2.5 2.5 0 1 1 .603 1.628l-6.718 3.12a2.499 2.499 0 0 1 0 1.504l6.718 3.12a2.5 2.5 0 1 1-.488.876l-6.718-3.12a2.5 2.5 0 1 1 0-3.256l6.718-3.12A2.5 2.5 0 0 1 11 2.5zm-8.5 4a1.5 1.5 0 1 0 0 3 1.5 1.5 0 0 0 0-3zm11 5.5a1.5 1.5 0 1 0 0 3 1.5 1.5 0 0 0 0-3z"
        />
      </svg>
      <span
        className="ms-2"
      >
        共有
      </span>
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      <svg
        className="align-text-bottom text-white"
        fill="currentColor"
        height={20}
        viewBox="0 0 16 16"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.921 11.9 1.353 8.62a.719.719 0 0 1 0-1.238L5.921 4.1A.716.716 0 0 1 7 4.719V6c1.5 0 6 0 7 8-2.5-4.5-7-4-7-4v1.281c0 .56-.606.898-1.079.62z"
        />
      </svg>
      <span
        className="ms-2 text-nowrap"
      >
        戻す
      </span>
    </button>
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      <svg
        className="align-text-bottom text-white"
        fill="currentColor"
        height={18}
        viewBox="0 0 16 16"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m.5 3 .04.87a1.99 1.99 0 0 0-.342 1.311l.637 7A2 2 0 0 0 2.826 14H9v-1H2.826a1 1 0 0 1-.995-.91l-.637-7A1 1 0 0 1 2.19 4h11.62a1 1 0 0 1 .996 1.09L14.54 8h1.005l.256-2.819A2 2 0 0 0 13.81 3H9.828a2 2 0 0 1-1.414-.586l-.828-.828A2 2 0 0 0 6.172 1H2.5a2 2 0 0 0-2 2zm5.672-1a1 1 0 0 1 .707.293L7.586 3H2.19c-.24 0-.47.042-.683.12L1.5 2.98a1 1 0 0 1 1-.98h3.672z"
        />
        <path
          d="M13.5 10a.5.5 0 0 1 .5.5V12h1.5a.5.5 0 1 1 0 1H14v1.5a.5.5 0 1 1-1 0V13h-1.5a.5.5 0 0 1 0-1H13v-1.5a.5.5 0 0 1 .5-.5z"
        />
      </svg>
      <span
        className="ms-2"
      >
        ディレクトリにページを追加する
      </span>
    </button>
  </div>
</div>
`;
